test/Main.hs

```diff
@@ -1590,6 +1590,74 @@ migrationTest5 connSource =
       rows :: [Maybe Bool] <- fetchMany runIdentity
       liftIO . assertEqual "All name_is_true are true" True $ all (== Just True) rows
 
+foreignKeyIndexesTests :: ConnectionSourceM (LogT IO) -> TestTree
+foreignKeyIndexesTests connSource =
+  testCaseSteps' "Foreign key indexes tests" connSource $ \step -> do
+    freshTestDB step
+
+    step "Create database with two tables, no foreign key checking"
+    do
+      let options = defaultExtrasOptions
+      migrateDatabase options ["pgcrypto"] [] [] [table1, table2]
+        [createTableMigration table1, createTableMigration table2]
+      checkDatabase defaultExtrasOptions [] [] [table1, table2]
+
+    step "Create database with two tables, with foreign key checking"
+    do
+      let options = defaultExtrasOptions { eoCheckForeignKeysIndexes = True }
+      assertException "Foreign keys are missing" $ migrateDatabase options ["pgcrypto"] [] [] [table1, table2]
+        [createTableMigration table1, createTableMigration table2]
+
+    step "Table is missing several foreign keys"
```

Comment:
I think you mean `is missing several indices` ?

Reply:
Oops, yes.